packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -505,7 +505,7 @@ class AppDomain extends Domain {
   }
 }
 
-typedef void _DeviceEventHandler(Device device);
+typedef Future<Null> _DeviceEventHandler(Device device);
```

Comment:
Actually, I wonder if it may not be necessary...

Take the following code:

```
Future<Null> foo(int i) async { print(i); }
void bar() {
  foo(1);
  foo(2);
  foo(3);
}
```

Does the VM guarantee that those will run in order (1, 2, 3)?  If it does, then you can leave the code as is.